### PR TITLE
Fix study and analysis objects/tests

### PR DESCRIPTION
**Pull request by josenavas**
It depends on #1529 so review/merge that one first.

Fixes the tests for the study and analysis objects. It also fixes the objects to retrieve instances of other objects rather than just the id.

### CODE REVIEW

**Comment by ElDeveloper** [comment]
@josenavas, when you have a chance can you pull from `upstream artifact`?

**Comment by josenavas** [comment]
@ElDeveloper you should review/merge #1529 first.

File: qiita_db/analysis.py [hunk 1]
```diff
@@ -705,64 +709,64 @@ def add_samples(self, samples):
...
-        proc_data : list, optional
-            processed data ids to remove, default None
+        artifacts : list, optional
+            Artifacts to remove, default None
         samples : list, optional
             sample ids to remove, default None
 
         Notes
         -----
         When only a list of samples given, the samples will be removed from all
```
Comment (antgonza) [inline qiita_db/analysis.py]: I know you didn't change this but are the black lines necessary? In the past we have use - at the beginning of each paragraph. Obviously, not blocking.

File: qiita_db/analysis.py [hunk 2]
```diff
@@ -705,64 +709,64 @@ def add_samples(self, samples):
...
         with qdb.sql_connection.TRN:
             self._lock_check()
-            if proc_data and samples:
+            if artifacts and samples:
                 sql = """DELETE FROM qiita.analysis_sample
                          WHERE analysis_id = %s
-                            AND processed_data_id = %s
+                            AND artifact_id = %s
                             AND sample_id = %s"""
                 # build tuples for what samples to remove from what
```
Comment (antgonza) [inline qiita_db/analysis.py]: This comment doesn't make sense, could you rewrite?

File: qiita_db/analysis.py [hunk 3]
```diff
@@ -809,13 +813,13 @@ def build_files(self, rarefaction_depth=None):
             self._build_biom_tables(samples, rarefaction_depth)
 
     def _get_samples(self):
-        """Retrieves dict of samples to proc_data_id for the analysis"""
+        """Retrieves dict of samples to artifact_id for the analysis"""
```
Comment (antgonza) [inline qiita_db/analysis.py]: Could you rewrite?

File: qiita_db/study.py [hunk 4]
```diff
@@ -671,20 +699,18 @@ def investigation(self):
             inv = qdb.sql_connection.TRN.execute_fetchindex()
             # If this study belongs to an investigation it will be in
             # the first value of the first row [0][0]
-            return inv[0][0] if inv else None
+            return qdb.investigation.Investigation(inv[0][0]) if inv else None
 
     @property
     def sample_template(self):
-        """ Returns sample_template information id
+        """ Returns sample_template information
```
Comment (ElDeveloper) [inline qiita_db/study.py]: Can you remove the leading space in this comment?

**Comment by ElDeveloper** [comment]
:+1: looks good to me!

File: qiita_db/analysis.py (hunk 1 above)
Comment (josenavas) [inline qiita_db/analysis.py]: Done

File: qiita_db/analysis.py (hunk 2 above)
Comment (josenavas) [inline qiita_db/analysis.py]: Done

File: qiita_db/analysis.py (hunk 3 above)
Comment (josenavas) [inline qiita_db/analysis.py]: Done

File: qiita_db/study.py (hunk 4 above)
Comment (josenavas) [inline qiita_db/study.py]: Done

**Pull request merged**